bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ThingActionsResource.java

```diff
@@ -166,13 +171,12 @@ public Response getActions(@PathParam("thingUID") @Parameter(description = "thin
             ThingActions thingActions = thingActionsEntry.getValue();
```

Comment:
For performance reasons I would suggest to split this:

- the first part of the loop (l. 174 to 159) which is time-consuming and static (after the methods of a given `ThingActions` will never change) should be done when the `ThingActions` are added (and stored in something that allows to remove them when the `ThingActions`  is removed)
- the second part (checking of action type and condig description) should take place here as it depends on the availability of these in the registries (which is dynamic)

This can be refactored later if you find it too complicated now.